File: 1. Two Sum.cpp
//url = https://leetcode.com/problems/two-sum/
//1st solution
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int len = nums.size();
        vector<int>ans;
        for(int i=0;i<len-1;i++){
            for(int j=i+1;j<len;j++){
                if(nums[i]+nums[j]==target) {ans.push_back(i);
                ans.push_back(j);}
            }
        }
        return ans;
    }
        
};
///2nd solution
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int len = nums.size();
        vector<int>ans;
        vector<pair<int,int>>ar;
        for(int i = 0; i < len; i++)
        {
            ar.push_back({nums[i], i});
        }
        sort(ar.begin(),ar.end());
        bool flag=false;
        int i=0,j=len-1;
         
        while(i<j)
        {
            
            if(ar[i].first+ar[j].first==target)
            {
                ans.push_back(ar[i].second);
                ans.push_back(ar[j].second);
                return ans;
            }
            else if(ar[i].first+ar[j].first>target)
            {
                j--;
            }
            else
                i++;
        }
        
        return ans;
        
    }
        
};

File: JavaScript/1. Two Sum.js
// url = https://leetcode.com/problems/two-sum/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for(let x = 0; x<nums.length-1; x++){
        for(let y = x+1; y<nums.length; y++){
            // console.log(x,y);
            if(nums[x]+nums[y]==target) return [x,y];
        }
    }
    return [];
};
